src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -50,6 +50,15 @@ class AttributeMetadata implements AttributeMetadataInterface
      */
     public $serializedName;
 
+    /**
+     * @var array|null
+     *
+     * @internal This property is public in order to reduce the size of the
+     *           class' serialized representation. Do not access it. Use
+     *           {@link getEmbedProperties()} instead.
+     */
+    public $embedProperties = [];
```

Comment:
As by definition properties are uniques per class, you should use a map instead of an array (something like `['prop1' => true, 'prop2' => true]`). The operation to check if the attribute must be embedded (in the normalizer) will then be an O(1) operation instead of an O(n) currently (`isset($embedProps[$propName])` instead of the `foreach` loop or the `in_array` call).

Reply:
I did it on `addEmbedProperty`. 

However, in the normalization process, I still need to do a loop on all EmbedProperties to retrieve their values.

In the denormalization process, I need to browse all attributeMetadata for current object to find the attribute which will carry the embedProperty. Do you see a best way to find it ?